examples/run_Hahn_echo_mw.py

```diff
@@ -1,97 +1,97 @@
-'''Run an Enhancement Experiment
+"""Run an Enhancement Experiment
 ================================
 Uses power control server so this will need to be running in sync. To do so:
     1. Open Xepr on the EPR computer, connect to spectrometer, and enable XEPR_API.
     2. In a separate terminal on the EPR computer, run the program XEPR_API_server.py and wait for it to tell you 'I am listening'.
     3. On the NMR computer, open a separate terminal in git/inst_notebooks/Instruments and run winpty power_control_server(). When ready to go it will say 'I am listening'.
     4. run this program to collect data
-'''    
+"""
 from pyspecdata import *
 from numpy import *
 import os
 import sys
 import SpinCore_pp
-from Instruments import Bridge12,prologix_connection,gigatronics
+from SpinCore_pp.ppg import run_spin_echo
+from Instruments import Bridge12, prologix_connection, gigatronics,power_control
 from serial import Serial
 import time
 from datetime import datetime
 from SpinCore_pp.power_helper import gen_powerlist
 
 fl = figlist_var()
-# {{{ experimental parameters
-#{{{power settings
-max_power_W = 4 #W
-power_steps = 18
-dB_settings = gen_powerlist(max_power_W,power_steps)
-append_dB = [dB_settings[abs(10**(dB_settings/10.-3)-max_power_W*frac).argmin()]
-        for frac in [0.75,0.5,0.25]]
-dB_settings = append(dB_settings,append_dB)
-print("dB_settings",dB_settings)
-print("correspond to powers in Watts",10**(dB_settings/10.-3))
+# {{{importing acquisition parameters
+config_dict = SpinCore_pp.configuration("active.ini")
+nPoints = int(config_dict["acq_time_ms"] * config_dict["SW_kHz"] + 0.5)
+# }}}
+# {{{create filename and save to config file
+date = datetime.now().strftime("%y%m%d")
+config_dict["type"] = "enhancement"
+config_dict["date"] = date
+config_dict["odnp_counter"] += 1
+filename = f"{config_dict['date']}_{config_dict['chemical']}_{config_dict['type']}_{config_dict['odnp_counter']}"
+# }}}
+# {{{power settings
+dB_settings = gen_powerlist(
+    config_dict["max_power"], config_dict["power_steps"] + 1, three_down=True
+)
+print("dB_settings", dB_settings)
+print("correspond to powers in Watts", 10 ** (dB_settings / 10.0 - 3))
 input("Look ok?")
-powers = 1e-3*10**(dB_settings/10.)
-#}}}
-date = datetime.now().strftime('%y%m%d')
-output_name = 'TEMPOL_289uM_heat_exch_0C'
-node_name = 'enhancement'
-adcOffset = 28
-carrierFreq_MHz = 14.549013
-nScans = 1
-nEchoes = 1
+powers = 1e-3 * 10 ** (dB_settings / 10.0)
+# }}}
+#{{{phase cycling and checking nPoints
 phase_cycling = True
 if phase_cycling:
     nPhaseSteps = 4
```

Comment:
although this section is obsolete (it's the default phase cycling for echo.py), this is used later for chunking the data before saving